myadmin-service-impls/student-services/src/main/java/za/ac/unisa/myadmin/student/services/decorators/ModuleEnrolmentServiceValidationDecorator.java

```diff
@@ -24,25 +33,130 @@
 
 	private StudentService studentService;
 
-	public void setStudentService(StudentService studentService) {
-		this.studentService = studentService;
-	}
-
 	@Override
 	public List<ModuleEnrolmentInfo> requestStudentModuleEnrolments(StudentInfo student)
-			throws MissingParameterException, InvalidParameterException, OperationFailedException,
-			DoesNotExistException {
-		if (!StringUtils.hasText(String.valueOf(student.getStudentNumber()))) {
-			throw new OperationFailedException("Please enter the student number.");
+		throws MissingParameterException, InvalidParameterException, OperationFailedException,
+		DoesNotExistException {
+
+		validateStudentExists(student);
+		validateStudentContactDetail(student);
+
+		return getNextDecorator().requestStudentModuleEnrolments(student);
+	}
+
+	/**
+	 *
+	 * @param student
+	 * @return
+	 * @throws PropertyVetoException
+	 * @throws OperationFailedException
+	 */
+	private StudentInfo executeProxy(StudentInfo student) throws PropertyVetoException, OperationFailedException {
+		final AtomicReference<OperationFailedException> exceptionReference = new AtomicReference<>();
+		final ActionListener exceptionListener = e -> exceptionReference
+			.set(new OperationFailedException(e.getActionCommand()));
+		StudentInfo proxyStudent = new StudentInfo();
+
+		// Get a reference to the proxy
+		final Srcds01sMntStudContactDetail studentContactDetailProxy = constructStudentContactDetailProxy();
+		studentContactDetailProxy.addExceptionListener(exceptionListener);
+		studentContactDetailProxy.setInWsStudentNr(student.getStudentNumber());
+		studentContactDetailProxy.execute();
+		String errorMsg = studentContactDetailProxy.getOutCsfStringsString500();
+
+		if (StringUtils.hasText(errorMsg)) {
```

Comment:
I've noticed that sometimes springframework.StringUtils is been used and other times common.lang.StringUtils. We should standardized on using the one or the other but not both.
I much prefer commons.lang.StringUtils only because I'm most familiar with it.